### PR TITLE
fix #8387 chore(cirrus): pyright type checker for cirrus

### LINKED ISSUE #8387: Set up Python type checker
We used mypy in experimenter, but we do have alternative available pytype [pytype](https://github.com/google/pytype), investigate which is better and based on pros and cons, use the type checker for cirrus service

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3216)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We want to add type checker for cirrus

This commit

- Adds a `pyright` dependency in `pyproject.toml`
- Include pyright in `make cirrus_check `command
- Make a command to create stubs for cirrus i.e. `make cirrus_typecheck_createstub`
- Added initially created a stub for cirrus in  `typings` folder

### CODE REVIEW

**Review by eliserichards — COMMENTED** [review]
This looks good to me, but I'm going to leave the approval to the others since I don't have much experience in this area 😅

**Comment by jeddai** [comment]
^ Ditto 😬

File: Makefile [hunk 1]
```diff
@@ -203,8 +203,8 @@ CIRRUS_BLACK_FIX = black -l 90 cirrus/server
 CIRRUS_RUFF_CHECK = ruff cirrus/server
 CIRRUS_RUFF_FIX = ruff --fix cirrus/server
 CIRRUS_PYTEST = pytest cirrus/server --cov=cirrus
-
-
+CIRRUS_PYTHON_TYPECHECK = pyright -p cirrus/server
+CIRRUS_PYTHON_TYPECHECK_CREATESTUB = pyright -p cirrus/server --createstub cirrus
```
Comment (jeddai) [inline Makefile:207]: Where do we plan to use this?

Comment (yashikakhurana) [reply Makefile:207]: I am using cirrus_python_typecheck in the make check command to verify the strict type, and the second command is to generate the stubs for the python code, I have ran it initially but later on can also be added a part of check command to make sure ours stubs are up to date with the code, right now just added as a make command

**Review by jeddai — APPROVED** [review]
Ah I see it's used in experimenter too. Looks good to me!

**Pull request merged**